src/orders/controller/orders.controller.ts

```diff
@@ -0,0 +1,127 @@
+import {
+  BadRequestException,
+  Body,
+  Controller,
+  Delete,
+  Get,
+  Param,
+  ParseUUIDPipe,
+  Post,
+  Put,
+} from '@nestjs/common';
+import OrdersMapper from '../mapper/orders.mapper';
+import { OrdersService } from '../service/orders.service';
+import { CustomersService } from 'src/customers/service/customers.service';
+import OrderDTO from '../dto/order.dto';
+import { ApiResponse } from '@nestjs/swagger';
+import { OrderDetailsService } from '../service/orderDetails.service';
+import OrderDetailsMapper from '../mapper/orderDetails.mapper';
+
+@Controller('orders')
+export class OrdersController {
+  constructor(
+    private ordersMapper: OrdersMapper,
+    private ordersService: OrdersService,
+    private customersService: CustomersService,
+    private orderDetailsService: OrderDetailsService,
+    private orderDetailsMapper: OrderDetailsMapper,
+  ) {}
+
+  @ApiResponse({ status: 200, description: 'Returns the orders' })
+  @Get()
+  async getOrders(): Promise<OrderDTO[]> {
+    const orders = await this.ordersService.getOrders();
+    const ordersDTO: OrderDTO[] = [];
+    orders.map((order) => {
+      const orderDetailDTO = this.orderDetailsMapper.orderDetailsToDtos(
+        order.orderDetails,
+      );
+      ordersDTO.push(this.ordersMapper.orderToDto(order, orderDetailDTO));
+    });
+    return ordersDTO;
+  }
+
+  @ApiResponse({ status: 200, description: 'Returns the order' })
+  @ApiResponse({ status: 404, description: "The order can't be found" })
+  @Get(':id')
+  async getOrderById(
+    @Param('id', ParseUUIDPipe) id: string,
+  ): Promise<OrderDTO | null> {
+    const order = await this.ordersService.getOrderById(id);
+    return this.ordersMapper.orderToDto(order, []);
+  }
+
+  @ApiResponse({
+    status: 201,
+    description: 'The order was created successfully',
+  })
+  @Post()
+  async createOrder(@Body() orderDTO: OrderDTO): Promise<OrderDTO> {
```

Comment:
In this method, you have written business logic (which should be at the service level). The controller should only call a delegated service method (like `createOrder`) and that delegated service method should include all the steps: `validateOrderDetails`, `getCustomerById`, `decreaseStock`, etc. 

The controller logic should be as simple as possible: just a few service calls and several mappings between DTOs and entities.

Reply:
done